src/deepsparse/server/build_logger.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Specifies the mapping from the ServerConfig to the DeepSparse Logger
+"""
+
+from typing import Any, Dict, List
+
+from deepsparse import BaseLogger, FunctionLogger, MultiLogger, PythonLogger
+from deepsparse.loggers.helpers import get_function_and_function_name
+from deepsparse.server.config import ServerConfig
+
+
+__all__ = ["build_logger"]
+
+_LOGGER_MAPPING = {"python": PythonLogger}
+
+
+def build_logger(server_config: ServerConfig) -> BaseLogger:
+    """
+    Builds a DeepSparse logger from the ServerConfig.
+
+    The process follows the following hierarchy:
+
+    First: if global logger config is provided,
+        the "leaf" loggers are built.
+
+    Second: if data logging config is specified, a set of
+        function loggers wraps around the appropriate "leaf" loggers.
+
+    Third: if required, the list of resulting loggers is wrapped inside a
+        MultiLogger.
+
+    :param server_config: the Server configuration model
+    :return: a DeepSparse logger instance
+    """
+
+    loggers_config = server_config.loggers
+    if not loggers_config:
+        return None
+    leaf_loggers = build_leaf_loggers(loggers_config)
+
+    loggers = build_function_loggers(server_config.endpoints, leaf_loggers)
+    if not loggers:
```

Comment:
we need to think about this more - this would mean that ALL data gets logged which we don't want in a default case. Instead should just be logging timings here.

We should add some base function loggers to log timings and have those wrap the default loggers which would prevent this case. Can do in follow up if necessary, but let's remove this case before landing